src/modules/endpoint/sagas/persistCategories.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+
+import type { Saga } from 'redux-saga'
+import MemoryDatabase from '../MemoryDatabase'
+import { CategoryModel } from '@integreat-app/integreat-api-client'
+import RNFetchblob from 'rn-fetch-blob'
+import { call } from 'redux-saga/effects'
+import { OFFLINE_CACHE_PATH } from '../../platform/constants/webview'
+import performanceNow from '../../app/performanceNow'
+
+const strMapToObj = strMap => {
+  const obj = Object.create(null)
+  for (const [k, v] of strMap) {
+    // We don’t escape the key '__proto__'
+    // which can cause problems on older engines
+    obj[k] = v
+  }
+  return obj
+}
+
+export default function * persistCategories (database: MemoryDatabase): Saga<void> {
```

Comment:
This does not persist but load from the db, right?

Reply:
It uses the data from the memory db to write to a file. This is not the final design yet.